### DISCUSSION TITLE
QoS for Joint State Monitoring

### DISCUSSION THREAD

**Discussion by acbuynak** [tesseract-robotics/tesseract_ros2#39] [Q&A] [answered]
Not really an Issue, but a discussion question.

I noticed that the joint_state monitoring is coded in as `SensorData` (best effort & volatile) which I think makes sense with it being a hardware sensor.. versus say the default QoS. But I can also see the case where it would be advantageous to let the subscriber QoS be automatically picked based on the upstream or at least parameterize this for dev's.

https://github.com/tesseract-robotics/tesseract_ros2/blob/b16609cee8b34bbcab5977a228f9be9bafed75ce/tesseract_monitoring/src/current_state_monitor.cpp#L112-L116

I found myself manually changing this today to work with a joint_state publisher I had set to a Reliable durability, hence the note. Obligatory link to [QoS Concepts in the ROS docs](https://docs.ros.org/en/rolling/Concepts/About-Quality-of-Service-Settings.html). 

Feel free to let me know if I'm totally off base here.

**Comment by gavanderhoorn** [comment, marked as answer]
> at least parameterize this for dev's.

I can't find the API / docs for it right now, but ROS 2 has support for QoS overrides which don't require source-level changes.

The subscriber should be created with a special, sentinel, QoS profile and the rest (ie: CLA parsing, forwarding the settings to the subscriber ctor, etc) should be taken care of.
